src/qgis_plugin_dev_tools/build/rewrite_imports.py

```diff
@@ -115,10 +116,33 @@ def rewrite_imports_in_source_file(
         flags=re.M,
     )
 
-    # add a note describing changes made to files
-    change_note = "# original source code changed by rewriting import statements\n"
-    if change_note not in contents:
-        contents = change_note + contents
+    # special case for custom widgets in .ui files
+    if source_file.suffix == ".ui":
+        ui_tree = ElementTree.fromstring(contents)  # noqa: SC200
+        for widget_section in ui_tree.iter("customwidget"):
+            header_section = widget_section.find("header")
+
+            if header_section is None or header_section.text is None:
+                continue
+
+            if header_section.text == rewritten_package_name:
+                header_section.text = header_section.text.replace(
```

Comment:
this could be simplified as `header_section.text = f"{container_package_name}.{rewritten_package_name}"` without replace